cognitive_architecture/database/create_database.py

```diff
@@ -1,63 +1,135 @@
+        if self.db_type in ["sqlite", "duckdb"]:
+            # For SQLite and DuckDB, check if the database file exists
+            return os.path.exists(db_name)
+        else:
+            query = text(f"SELECT 1 FROM pg_database WHERE datname='{db_name}'")
+            async with self.get_connection() as connection:
+                result = await connection.execute(query)
+                return await result.fetchone() is not None
 
-def create_tables(engine):
-    Base.metadata.create_all(bind = engine)
+    async def create_database(self, db_name):
+        """Create a new database."""
+        if self.db_type not in ["sqlite", "duckdb"]:
+            # For databases like PostgreSQL, create the database explicitly
+            async with self.get_connection() as connection:
+                await connection.execute(text(f"CREATE DATABASE {db_name}"))
 
-if __name__ == "__main__":
-    host = os.environ.get('POSTGRES_HOST')
-    username = os.environ.get('POSTGRES_USER')
-    password = os.environ.get('POSTGRES_PASSWORD')
-    database_name = os.environ.get('POSTGRES_DB')
+    async def drop_database(self, db_name):
+        """Drop an existing database."""
+        if self.db_type in ["sqlite", "duckdb"]:
+            # For SQLite and DuckDB, simply remove the database file
+            os.remove(db_name)
+        else:
+            async with self.get_connection() as connection:
+                await connection.execute(text(f"DROP DATABASE IF EXISTS {db_name}"))
+
+    async def create_tables(self):
+        """Create tables based on the SQLAlchemy Base metadata."""
+        async with self.engine.begin() as conn:
+            await conn.run_sync(Base.metadata.create_all)
 
-    engine = create_admin_engine(username, password, host, database_name)
-    connection = engine.connect()
+if __name__ == "__main__":
+    async def main():
+        """Runs as a part of startup docker scripts to create the database and tables."""
 
-    # print(Base.metadata.tables)
+        dbconfig = DatabaseConfig(db_type=config.db_type, db_name=config.db_name)
+        db_manager = DatabaseManager(config=dbconfig)
+        database_name = dbconfig.db_name
 
-    if not database_exists(connection, database_name):
-        logger.info(f"Database {database_name} does not exist. Creating...")
-        create_database(connection, database_name)
-        logger.info(f"Database {database_name} created successfully.")
+        if not await db_manager.database_exists(database_name):
+            print(f"Database {database_name} does not exist. Creating...")
+            await db_manager.create_database(database_name)
+            print(f"Database {database_name} created successfully.")
 
-    connection.close()
-    engine.dispose()
+        await db_manager.create_tables()
 
-    create_tables(engine)
\ No newline at end of file
+    asyncio.run(main())
+#
+# def create_admin_engine(username, password, host, database_name):
```

Comment:
Let's delete the code we don't need, we have it in git history already.

Reply:
Refactored